src/sp/it/pl/core/CoreSerializer.kt

```diff
@@ -66,17 +68,20 @@ object CoreSerializer: Core {
      *
      */
     @Blocks
-    inline fun <reified T: Serializable> writeSingleStorage(o: T) {
-        val f = APP.DIR_LIBRARY.childOf(T::class.simpleName ?: T::class.jvmName)
-        try {
-            FileOutputStream(f).use {
-                ObjectOutputStream(it).use {
-                    it.writeObject(o)
+    inline fun <reified T: Serializable> writeSingleStorage(o: T): Try<Unit,Exception> {
+        val f = APP.DIR_LIBRARY/(T::class.simpleName ?: T::class.jvmName)
+        return f.writeSafely {
+            try {
+                FileOutputStream(it).use {
+                    ObjectOutputStream(it).use {
```

Comment:
I'm not sure I understand.

Reply:
Whenever a `use` block completes, it closes the stream, that is the main purpose, to be safe. But closing a stream is a cascading action, i.e. closing a stream will automatically close all streams created from it. If you have to `use` blocks based on the same stream, it will attempt to close it twice.

Reply:
I see.  I will look into hat is the proper implementation here.

Reply:
The proper implementation is to have a single `use` block on the outermost stream. Since the ObjectOutputStream simply wraps the FileOutputStream, there is no need to close it when the FileOutputStream already takes care of that.